### PR TITLE
added MPI dependency to Nekbone package

### PR DESCRIPTION
**Pull request by pkondamudi**

### CODE REVIEW

File: var/spack/repos/builtin/packages/nekbone/package.py [hunk 1]
```diff
@@ -44,7 +51,10 @@ def install(self, spec, prefix):
 
         for wdir in working_dirs:
             with working_dir('test/' + wdir):
-                makenek = Executable('./makenek')
+                makenec = FileFilter('makenek')
+                makenec.filter('CC.*', 'CC='+self.spec['mpi'].mpicc)
+                makenec.filter('FF77.*', 'FF77='+self.spec['mpi'].mpif77)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nekbone/package.py]: You need to check whether or not `+mpi` is in the spec before using this.

File: var/spack/repos/builtin/packages/nekbone/package.py [hunk 2]
```diff
@@ -44,7 +51,10 @@ def install(self, spec, prefix):
...
+                makenek = Executable('./makenek~')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nekbone/package.py]: This is the backup of the original file, without the patches. Is this what you want?

Comment (pkondamudi) [inline var/spack/repos/builtin/packages/nekbone/package.py]: Yes.

File: var/spack/repos/builtin/packages/nekbone/package.py (hunk 1 above)
Comment (pkondamudi) [inline var/spack/repos/builtin/packages/nekbone/package.py]: I want `+mpi` to be there always.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nekbone/package.py]: So then get rid of the variants and make it always depend on MPI?

File: var/spack/repos/builtin/packages/nekbone/package.py (hunk 2 above)
Comment (pkondamudi) [inline var/spack/repos/builtin/packages/nekbone/package.py]: Changed pointer to the filtered file. The filtered changes are in actual file.

**Pull request merged**